### ISSUE TITLE
Code integrity check shows extra .htaccess file in apparently empty tmp folder

### ISSUE THREAD

**Issue by bpcurse** [nextcloud/server#7132]
### Steps to reproduce
Unclear

Two events were time related:
- Failure to install Deck 0.2.5 (maintenance mode had to be disabled manually to start update)
- Deleted core/skeleton files what triggered an integrity check problem. Restored these files from another 12.0.3 install. Set up custom skeleton directory outside the nextcloud install path.

### Expected behaviour
Integrity Check should throw no error

### Actual behaviour
Integrity Check reports extra file "tmp/oc_tmp_KxFUQ4-folder/.htaccess" but this subdirectory and .htaccess file obviously do not exist.
Changing the tmp directory doesn't help -> an extra .htaccess file is reported again, this time in the new temporary directory path.

### Server configuration
Shared Webhosting (all-inkl.com)

**Nextcloud version:** (see Nextcloud admin page)
12.0.3

**Updated from an older Nextcloud/ownCloud or fresh install:**
Updated from 12.0.2 a while ago (no problems then)

**Where did you install Nextcloud from:**
Software installer script provided by webhoster

```
Technical information
=====================
The following list covers which files have failed the integrity check. Please read
the previous linked documentation to learn more about the errors and how to fix
them.

Results
=======
- core
	- EXTRA_FILE
		- tmp/oc_tmp_KxFUQ4-folder/.htaccess

Raw output
==========
Array
(
    [core] => Array
        (
            [EXTRA_FILE] => Array
                (
                    [tmp/oc_tmp_KxFUQ4-folder/.htaccess] => Array
                        (
                            [expected] => 
                            [current] => 59fae9bd43c1a7754899d925ca1138b99356100f160a3e491094b27401934750ef2a8a43ca8486e1007bdf674aa5bd45df9067dd050380672a29985a3ddf3887
                        )

                )

        )

)
```

**List of activated apps:**
<details>
<summary>App list</summary>

Enabled:
  - activity: 2.5.2
  - announcementcenter: 3.1.0
  - apporder: 0.4.0
  - audioplayer: 2.1.0
  - bookmarks: 0.10.1
  - bruteforcesettings: 1.0.2
  - calendar: 1.5.6
  - checksum: 0.3.5
  - comments: 1.2.0
  - contacts: 2.0.1
  - dav: 1.3.0
  - deck: 0.2.6
  - drawio: 0.8.8
  - drop_account: 0.0.6
  - external: 2.0.3
  - federatedfilesharing: 1.2.0
  - federation: 1.2.0
  - files: 1.7.2
  - files_accesscontrol: 1.2.5
  - files_clipboard: 0.6.4
  - files_downloadactivity: 1.1.1
  - files_external: 1.3.0
  - files_markdown: 2.0.1
  - files_pdfviewer: 1.1.1
  - files_sharing: 1.4.0
  - files_texteditor: 2.4.1
  - files_trashbin: 1.2.0
  - files_versions: 1.5.0
  - files_videoplayer: 1.1.0
  - firstrunwizard: 2.1
  - gallery: 17.0.0
  - groupfolders: 1.1.0
  - keeweb: 0.4.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - mail: 0.7.4
  - metadata: 0.6.0
  - nextcloud_announcements: 1.1
  - notes: 2.3.1
  - notifications: 2.0.0
  - oauth2: 1.0.5
  - password_policy: 1.2.2
  - polls: 0.8.0
  - provisioning_api: 1.2.0
  - ransomware_protection: 1.0.5
  - serverinfo: 1.2.0
  - sharebymail: 1.2.0
  - socialsharing_email: 1.0.1
  - socialsharing_facebook: 1.0.1
  - survey_client: 1.0.0
  - systemtags: 1.2.0
  - tasks: 0.9.5
  - theming: 1.3.0
  - twofactor_backupcodes: 1.1.1
  - twofactor_totp: 1.3.1
  - updatenotification: 1.2.0
  - user_ldap: 1.2.1
  - workflowengine: 1.2.0

Disabled:
  - admin_audit
  - circles
  - dashboard
  - encryption
  - music
  - ojsxc
  - spreed
  - user_external
  - workin2gether

</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

{
    "system": {
        "instanceid": "***REMOVED***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED***"
        ],
        "datadirectory": "\/www\/htdocs\/***REMOVED***\/cloud\/data",
        "overwrite.cli.url": "***REMOVED***",
        "dbtype": "mysql",
        "version": "12.0.3.3",
        "dbname": "***REMOVED***",
        "dbhost": "localhost",
        "dbport": "",
        "dbtableprefix": "LnNIq_",
        "mysql.utf8mb4": false,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "tempdirectory": "\/www\/htdocs\/***REMOVED***\/cloud\/tmp\/",
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "\\OCA\\User_LDAP\\LDAPProviderFactory",
        "mail_from_address": "cloud",
        "mail_smtpmode": "php",
        "mail_smtpauthtype": "LOGIN",
        "mail_domain": "***REMOVED***",
        "maintenance": false,
        "theme": "",
        "loglevel": 2,
        "skeletondirectory": "\/www\/htdocs\/***REMOVED***\/cloud_skeleton_***REMOVED***"
    }
}

</details>

**Are you using external storage, if yes which one:** webdav

**Are you using encryption:** no

**Comment by Bugsbane** [comment]
Often files with a file name beginning with a period, such as .htaccess don't show in file managers unless you explicitly tell them to show hidden files.

Likely, it *is* in there but your file manager just isn't showing it to you. See if you can find the option. It's usually somewhere like view > show hidden files/folders (Ctrl + . On KDE Dolphin, Ctrl + h on Gnome Nautilus, not sure on Mac / windows)

**Comment by bpcurse** [comment]
Thank you for the response. I had already checked that but forgotten to mention it above.
It is not visible via FTP (Filezilla with the option active to force it to show all hidden files) and via SSH "ls -a".

**Comment by PrinceCharles23** [comment]
exactly the same here!
my file/folder is called "tmp/oc_tmp_oXPV6o-folder/.htaccess"

any news on the issue?

**Comment by bpcurse** [comment]
Thanks for commenting, it's good to know I'm not the only one with this problem. I did not hear anything new and the problem is persistent. I hope maybe an update to the next version - as soon as it is released - will magically fix it...?
Are you on a shared webhosting service, too?

**Comment by PrinceCharles23** [comment]
I'm also with all-inkl.com ;-)

I did a series of updates (I do not use owncloud that excessively) - 9.x.y -> 9.v.w (forgot the versions) and then 9.v.w -> 10.0.3. After that i had some missing/additional file errors that I naturally fixed by adding/deleting the files (from the zip file downloaded from owncloud).

only error that persists is the aforementioned .htaccess file in that non-existing temp folder.

**Comment by Bugsbane** [comment]
Hmmm... interesting that the only two people to report this are both on the exact same shared host. Have you tried contacting the host for ideas @bpcurse / @PrinceCharles23 , just in case thy have any weird security settings or something getting in the way...

**Comment by PrinceCharles23** [comment]
nope did not contact the hosting company. but i guess there will not be much (valuable) response either! ;-)

**Comment by bpcurse** [comment]
Definitely interesting and could be related. I will try to contact them tomorrow and report it here.

**Comment by PrinceCharles23** [comment]
any news?

**Comment by bpcurse** [comment]
Sorry, I did not manage to contact them due to my workload.
But I just updated to NC 12.0.4 from 12.0.3 and that solves the problem :)

**Comment by PrinceCharles23** [comment]
NC?
well, i guess it will be fixed in the next version without us ever knowing what it was. will have to wait for that...

**Comment by bpcurse** [comment]
NC = Nextcloud (the stable branch).
Good point, I will try to avoid abbreviations next time.

**Comment by PrinceCharles23** [comment]
thanks for clarifying!
owncloud is still on the old version number ;-)

**Comment by MorrisJobke** [comment]
> But I just updated to NC 12.0.4 from 12.0.3 and that solves the problem :)
> ... or at least the error is not displayed any more.

Then we can close this I guess. Sorry for the trouble you had.

**Comment by bpcurse** [comment]
Just wanted to add this because I probably found the connection to the webhoster. It doesn't really explain the error message, but could be linked to it.

all-inkl.com puts a "tmp" directory in the main nextcloud directory and defines it in config.php as temp directory. This leads to update problems, as well. We always had to delete the tmp directory, update and afterwards create a new one. I kept wondering why this wasn't an already known and solved bug.

Solution: Create the directory somewhere else and point to it inside config.php, then delete the old one. I created a "tmp_cloud" directory on the same path where the nextcloud main folder is located and checked that it isn't accessible from the outside world.
